### PR TITLE
DNM: EXPERIMENTAL: Wip 13937 17857 oliva

### PR DESCRIPTION
**Pull request by dzafman**
This is just holding code from Alexandre Oliva until we have a chance to review it.  Just commit https://github.com/dzafman/ceph/commit/763fe9d65884f3d868896704cbaf7de5e1c1c3fd

Fixes: http://tracker.ceph.com/issues/18162

### CODE REVIEW

File: src/osd/ReplicatedPG.cc [hunk 1]
```diff
@@ -9334,8 +9334,8 @@ SnapSetContext *ReplicatedPG::get_snapset_context(
 	r = pgbackend->objects_get_attr(oid.get_head(), SS_ATTR, &bv);
       if (r < 0) {
 	// try _snapset
-      if (!(oid.is_snapdir() && !oid_existed))
-	r = pgbackend->objects_get_attr(oid.get_snapdir(), SS_ATTR, &bv);
+	if (!(oid.is_snapdir() && !oid_existed))
+	  r = pgbackend->objects_get_attr(oid.get_snapdir(), SS_ATTR, &bv);
```
Comment (amitkumar50) [inline src/osd/ReplicatedPG.cc]: @dzafman NIT tab left

File: src/osd/ReplicatedPG.cc [hunk 2]
```diff
@@ -10921,18 +10981,19 @@ int ReplicatedPG::recover_backfill(
...
-  for (set<pg_shard_t>::iterator i = backfill_targets.begin();
-       i != backfill_targets.end();
-       ++i) {
-    peer_backfill_info[*i].trim_to(
-      MAX_HOBJ(peer_info[*i].last_backfill, last_backfill_started,
-	       get_sort_bitwise()));
+  if (trim) {
+    dout(20) << "trimming backfill interval up to "
+	     << last_backfill_started << dendl;
+    for (set<pg_shard_t>::iterator i = backfill_targets.begin();
```
Comment (amitkumar50) [inline src/osd/ReplicatedPG.cc]: @dzafman not an mandate But in my opinion would be good to use for-each/range-based for loop

Comment (mattbenjamin) [inline src/osd/ReplicatedPG.cc]: @amitkumar50 dealers choice

**Comment by dzafman** [comment]
See ceph/ceph#18145

**Pull request closed without merging**